nx-os/poap/poap.py

```diff
@@ -72,13 +73,10 @@
 # parameters passed through environment:
 # TODO: use good old argv[] instead, using env is bad idea.
 # pid is used for temp file name: just use getpid() instead!
-# serial number should be gotten from "show version" or something!
 pid=""
 if os.environ.has_key('POAP_PID'):
     pid=os.environ['POAP_PID']
-serial_number=None
-if os.environ.has_key('POAP_SERIAL'):
-    serial_number=os.environ['POAP_SERIAL']
+serial_number=re.search("ID\ (.+?)\n", cli("show version | grep Board")).group(1)
```

Comment:
Can this be done without a regex?  Maybe using split and a slice?  If a regex is required please use the r'ID\ (.+?)\n' notation so it is handled as a regex.